datafusion/src/physical_plan/mod.rs

```diff
@@ -509,6 +514,76 @@ pub trait Accumulator: Send + Sync + Debug {
     fn evaluate(&self) -> Result<ScalarValue>;
 }
 
+///  Source represents where the data comes from.
+enum Source<R = Box<dyn Read + Send + Sync>> {
```

Comment:
Might this be better in a new module, e.g. sources? WDYT @alamb

Reply:
I think that would improve the code, but I also think it would be fine to move the code into its own module as a follow on PR, depending on @heymind 's preference

Reply:
Ok. A new module named `source.rs` is more clear.